models/action.go

```diff
@@ -469,9 +469,17 @@ func UpdateIssuesCommit(doer *User, repo *Repository, commits []*PushCommit) err
 				continue
 			}
 
-			if err = issue.ChangeStatus(doer, repo, true); err != nil {
+			// Check for dependencies, if there aren't any, close it
+			noDeps, err := IssueNoDependenciesLeft(issue)
```

Comment:
If `there are dependencies left` is made into an error it would make this code much cleaner.

Reply:
You mean it should return an error saying there are dependencies left instead of _not closing it_?

Reply:
No. I mean that it should return `ErrDepedenciesFound` if a dependency is found...

Reply:
As an error, not a bool? (Sorry if I'm notunderstanding you)

If I understood you correctly, `UpdateIssuesCommit()` should return an error if there are dependencies left? How would I show the user a message saying he needs to close all depending issues? I think its cleaner to have a bool and create an errormessage from this instead checking if there was an error and thenchecking for what error and if it is the right type, display a message to the user... 

I'm still pretty new to Go, maybe this is the right approach but to me it doesn't seem like one.

Reply:
It would look something like this:
```go
err := IssueNoDependenciesLeft(issue)
switch err {
  case ErrDependencyFound:
    // show error
  case nil:
    // issue.ChangeStatus(...)
  default:
    return err
}

Reply:
It now checks in `ChangeStatus()`, not here...
I solved the problem otherwise, this comment seems a bit out of date (this pr is open wayyy too long now). Here it is checking for keywords in commit messages which would close the issue. If it finds one of these, but the issue still has unsatisfied dependencies, the push should not fail, but the issue won't be closed. So showing a message to the user doesn't actually happens here 🤔